tests/templates/test_subroutines.py

```diff
@@ -800,3 +785,156 @@ def circuit(w_ph_0, w_ph_1, w_pphh):
         jac_A = circuit.jacobian((w_ph_0, w_ph_1, w_pphh), method="A")
         jac_F = circuit.jacobian((w_ph_0, w_ph_1, w_pphh), method="F")
         assert jac_A == pytest.approx(jac_F, abs=tol)
+
+class TestApproxTimeEvolution:
+    """Tests for the ApproxTimeEvolution template from the pennylane.templates.subroutine module."""
+
+    def test_hamiltonian_error(self):
+        """Tests if the correct error is thrown when `hamiltonian` is not a pennylane.Hamiltonian object"""
+
+        n_wires = 2
+        dev = qml.device('default.qubit', wires=n_wires)
+
+        hamiltonian = np.array([[1, 1], [1, 1]])
+
+        @qml.qnode(dev)
+        def circuit():
+            ApproxTimeEvolution(hamiltonian, 2, 3)
+            return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        with pytest.raises(ValueError) as info:
```

Comment:
A minor comment, but would recommend using the patter `with pytest.raises(ValueError, match="..."):` to match the rest of the tests 🙂